src/workloads/execution/ColumnstoreIndexes.yml

```diff
@@ -0,0 +1,467 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
+Description: |
+  This workload compares the query performance with columnar indexes to performances without them.
+
+  The workload consists of the following phases and actors:
+    0. Warm up cache.
+    1. Run queries with columnstore indexes.
+    2. Drop columnstore indexes.
+    3. Warm up cache again for regular scans.
+    4. Run queries without columnstore indexes.
+
+Keywords:
+- columnstore
+- analytics
+
+# Defines the database name.
+Database: &db charts-metrics
+
+# Defines the collection name.
+Collection: &Collection events
+
+DateFieldProjectionQuery: &DateFieldProjectionQuery
+  aggregate: *Collection
+  pipeline:
+    [{"$project": {"_id": 0, "created_at": 1}}]
+  allowDiskUse: true
+  cursor: {batchSize: 2000}
+
+DistinctStringFieldProjectionQuery: &DistinctStringFieldProjectionQuery
+  aggregate: *Collection
+  pipeline:
+    [{"$project": {"_id": 0, "metadata.tenant_id": 1}}]
+  allowDiskUse: true
+  cursor: {batchSize: 2000}
+
+DuplicatedStringFieldProjectionQuery: &DuplicatedStringFieldProjectionQuery
+  aggregate: *Collection
+  pipeline:
+    [{"$project": {"_id": 0, "action": 1}}]
+  allowDiskUse: true
+  cursor: {batchSize: 2000}
+
+CountPerActionQuery: &CountPerActionQuery
+  aggregate: *Collection
+  pipeline:
+    [{"$group": {"_id": "$action", o: {$sum: 1}}}]
+  allowDiskUse: true
+  cursor: {batchSize: 2000}
+
+TotalTenantCreatedQuery: &TotalTenantCreatedQuery
+  aggregate: *Collection
+  pipeline:
+    [
+      {"$match": {"metadata.target": "cloud"}},
+      {"$group": {"_id": {}, "__alias_0": {"$addToSet": "$metadata.tenant_id"}}},
+      {"$project": {"_id": 0, "__alias_0": {"$size": "$__alias_0"}}},
+      {"$project": {"value": "$__alias_0", "_id": 0}},
+      {"$limit": 5000}
+    ]
+  allowDiskUse: true
+  cursor: {batchSize: 2000}
+
+ActiveCloudUsersLastMonthQuery: &ActiveCloudUsersLastMonthQuery
+  aggregate: *Collection
+  pipeline:
+    [
+      {"$addFields": {"target": 3000}},
+      {
+        "$match": {
+          "metadata.target": {"$in": ["cloud"]},
+          "created_at": {"$gte": {"$date": "2020-03-09T01:10:04Z"}},
```

Comment:
$date isn't actually an expression, so I think this is doing comparisons with the object literal `{$date: "blah blah"}` which prevents it from returning any results. Is there a way to check that the server returns X (or even greater than 0) results from the query?

Reply:
I didn't follow this, what is the problem with comparing with object literal `{"$date": "2020-03-09T01:10:04Z"}`?

Reply:
@borosaurus, you're right. It should be `{"$gte": ISODate("2020-03-09T01:10:04Z")}`. I don't think that there's $date operator. Instead, we have $dateFromString. Fixed.

Reply:
Yeah, I wish the behavior there would be different too. What's even worse is that the shell will sometimes print values using the `$date` syntax.

Anyway, one last question: is there any way to do this check for number of results within genny, rather than doing it manually?

Reply:
I don't know. It looks like that Alya is adding the assertive actor as part of TPC-H benchmarks:

https://github.com/mongodb/genny/pull/674

BTW, all queries are copied and pasted from https://docs.google.com/document/d/12Q_-w1PYDJD8nkRIXKSkGGSjJVsSn6JWrC3_wXl51vc/edit#

Better to update those queries too. I have neither comment access nor edit access to the doc.

Why are you asking? Is it a suggestion or a pure question out of curiosity?

Reply:
I'm just asking because if we change this benchmark in the future I don't want it to break without our knowing. But if there's no way to do basic correctness tests here, there's not much to be done :/

Reply:
I see. It sounds like a suggestion then. That's a valid concern and it's a current limitation of Genny. As I replied in the previous comment, Alya is adding an assertive actor and I think the actor can be used for result verification. Will wait for Alya's PR to be merged and add result verification if this PR is not urgent. LMK if that's desirable.

Reply:
Up to you. That PR is 19 days old so it may take awhile. We can always merge this and add the assertion later.

Reply:
ok, then will merge this and add verification later.